File: src/algorithm_java/leetcode/Problem406.java
package algorithm.leetcode;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.Comparator;
import java.util.List;

public class Problem406 {

    public static int[][] reconstructQueue(int[][] people) {
        Arrays.sort(people, new Comparator<int[]>() {
            @Override
            public int compare(int[] o1, int[] o2) {
                if (o1[0] < o2[0])
                    return 1;
                else if (o1[0] > o2[0])
                    return -1;
                else if (o1[1] < o2[1])
                    return -1;
                else if (o1[1] > o2[1])
                    return 1;
                return 0;
            }
        });
        List<int[]> tmp = new ArrayList<>();
        for (int i = 0; i < people.length; i++) {
            tmp.add(people[i][1], people[i]);
        }
        int[][] ans = new int[people.length][2];
        for (int i = 0; i < people.length; i++) {
            ans[i][0] = tmp.get(i)[0];
            ans[i][1] = tmp.get(i)[1];
        }
        return ans;
    }

    public static void main(String[] args) {
        int[][] people = new int[][] {
                {7,0},
                {4,4},
                {7,1},
                {5,0},
                {6,1},
                {5,2}
        };
        reconstructQueue(people);
    }

}

File: src/algorithm_cpp/problem406.cpp
class Solution {
public:
    vector<vector<int>> reconstructQueue(vector<vector<int>>& people) {
	vector<vector<int>> rt;
	sort(people.begin(), people.end(), [](const vector<int>& A, const vector<int>& B) -> int {
	    if (A[0] != B[0]) return A[0] > B[0];
	    else return A[1] < B[1];
	});
    
	for (int i = 0; i < people.size(); i++) {
        //cout << people[i][0] << "," << people[i][1] << endl;
	    rt.insert(rt.begin() + people[i][1], people[i]);
	}
	return rt;
    }
};

